internal/datastore/discussion.go

```diff
@@ -105,103 +107,50 @@ func (d *delphisDB) UpsertDiscussion(ctx context.Context, discussion model.Discu
-// 		if err != nil {
-// 			logrus.WithError(err).Warnf("ListDiscussion: Failed unmarshaling discussion: %+v", elem)
-// 			continue
-// 		}
-// 		edges = append(edges, &model.DiscussionsEdge{
-// 			Node: &discussionObj,
-// 		})
-// 		ids = append(ids, discussionObj.ID)
-// 	}
-
-// 	return &model.DiscussionsConnection{
-// 		IDs:   ids,
-// 		Edges: edges,
-// 	}, nil
-// }
-
-// func (d *delphisDB) GetDiscussionsByIDsDynamo(ctx context.Context, ids []string) (map[string]*model.Discussion, error) {
-// 	logrus.Debug("GetDiscussionsByIDs: Dynamo BatchGetItem")
-// 	keys := make([]map[string]*dynamodb.AttributeValue, 0)
-// 	for _, id := range ids {
-// 		keys = append(keys, map[string]*dynamodb.AttributeValue{
-// 			"ID": {
-// 				S: aws.String(id),
-// 			},
-// 		})
-// 	}
-// 	// NOTE: This has to be BatchGet because query will not work unless we specify partition key.
-// 	res, err := d.dynamo.BatchGetItem(&dynamodb.BatchGetItemInput{
-// 		RequestItems: map[string]*dynamodb.KeysAndAttributes{
-// 			d.dbConfig.Discussions.TableName: {
-// 				Keys: keys,
-// 			},
-// 		},
-// 	})
-
-// 	if err != nil {
-// 		logrus.WithError(err).Errorf("GetDiscussionsByIDs: Failed to retrive discussions with ids: %+v", ids)
-// 		return nil, err
-// 	}
-
-// 	discussionMap := map[string]*model.Discussion{}
-// 	for _, id := range ids {
-// 		discussionMap[id] = nil
-// 	}
-// 	elems := res.Responses[d.dbConfig.Discussions.TableName]
-// 	for _, elem := range elems {
-// 		discussionObj := model.Discussion{}
-// 		err = dynamodbattribute.UnmarshalMap(elem, &discussionObj)
-// 		if err != nil {
-// 			logrus.WithError(err).Warnf("GetDiscussionsByIDs: Failed to unmarshal discussion object: %+v", elem)
-// 			continue
-// 		}
-
-// 		discussionMap[discussionObj.ID] = &discussionObj
-// 	}
-
-// 	return discussionMap, nil
-// }
+func (d *delphisDB) GetDiscussionTags(ctx context.Context, id string) TagIter {
+	logrus.Debug("GetDiscussionTags::SQL Query")
+	if err := d.initializeStatements(ctx); err != nil {
+		logrus.WithError(err).Error("GetDiscussionTags::failed to initialize statements")
+		return &tagIter{err: err}
+	}
+
+	rows, err := d.prepStmts.getDiscussionTagsStmt.QueryContext(
+		ctx,
+		id,
+	)
+	if err != nil {
+		logrus.WithError(err).Error("failed to query GetDiscussionTags")
+		return &tagIter{err: err}
+	}
+
+	return &tagIter{
+		ctx:  ctx,
+		rows: rows,
+	}
+}
+
+func (d *delphisDB) PutDiscussionTags(ctx context.Context, tx *sql.Tx, tag model.Tag) (*model.Tag, error) {
+	logrus.Debug("PutDiscussionTags::SQL Create")
+	if err := d.initializeStatements(ctx); err != nil {
+		logrus.WithError(err).Error("PutDiscussionTags::failed to initialize statements")
+		return nil, err
+	}
+
+	if err := tx.StmtContext(ctx, d.prepStmts.putDiscussionTagsStmt).QueryRowContext(
+		ctx,
+		tag.ID,
+		tag.Tag,
+	).Scan(
+		&tag.ID,
+		&tag.Tag,
+		&tag.CreatedAt,
+	); err != nil {
+		if err == sql.ErrNoRows {
+			return &model.Tag{}, nil
```

Comment:
how do we want to handle collisions? this returns null on collisions.

Reply:
In this case a collision would be if a discussion already has the tags that are being added?